### PR TITLE
[ORA-618] Emissions Module Params as Keeper Stored Values Editable by Admin Whitelist Key

### PR DESCRIPTION
**Pull request by relyt29**
Deletes various constants in the code and replaces them with editable Params to the emissions module that are stored in the keeper and editable via an `UpdateParams` transaction

this PR lacks unit tests. Get to that on Monday.

### CODE REVIEW

**Review by kpeluso — COMMENTED** [review]
there's 1 thing you could that'd be safer

File: x/emissions/proto/emissions/state/v1/types.proto [hunk 1]
```diff
@@ -10,7 +10,39 @@ import "gogoproto/gogo.proto";
 
 // Params defines the parameters of the module.
 message Params {
-  string version = 1;
+  string version = 1; // version of the protocol should be in lockstep with github release tag version
+	int64 epoch_length = 2; // length of an "epoch" for rewards payouts in blocks
```
[the remark this answers is not in the document]
Comment (relyt29) [reply x/emissions/proto/emissions/state/v1/types.proto:14]: yes absolutely

File: x/emissions/keeper/keeper.go [hunk 2]
```diff
@@ -34,21 +34,6 @@ type BLOCK_NUMBER = int64
 type UNIX_TIMESTAMP = uint64
 type REQUEST_ID = string
 
-// Emissions rate Constants
-// TODO make these not constants and figure out how they should
-// be changeable by governance or some algorithm or whatever.
-// Likely should be params within the keeper.
-const EPOCH_LENGTH = 5
```
Comment (kpeluso) [inline x/emissions/keeper/keeper.go:41]: also, think these need to be initialized to something (perhaps just what's here) at genesis, right?

Comment (relyt29) [reply x/emissions/keeper/keeper.go:41]: they are

File: x/emissions/module/autocli.go [hunk 3]
```diff
@@ -98,10 +98,15 @@ func (am AppModule) AutoCLIOptions() *autocliv1.ModuleOptions {
...
-				// 	RpcMethod: "UpdateParams",
-				// 	Skip:      true, // This is a authority gated tx, so we skip it.
-				// },
+				{
+					RpcMethod: "UpdateParams",
+					Use:       "update-params [sender] [params]",
+					Short:     "Add a new topic to the network",
+					PositionalArgs: []*autocliv1.PositionalArgDescriptor{
+						{ProtoField: "sender"},
+						{ProtoField: "params"},
```
[the remark this answers is not in the document]
Comment (relyt29) [reply x/emissions/module/autocli.go:107]:
```bash
allorad tx emissions update-params --generate-only "alice" '{"version":["v0.0.4"], "epoch_length":[5], "emissions_per_epoch":["1000"], "min_topic_unmet_demand":["100"], "max_topics_per_block":[1000], "min_request_unmet_demand":["1"], "max_allowable_missing_inference_percent":[10], "required_minimum_stake":["1"], "remove_stake_delay_window":[172800], "min_fastest_allowed_cadence":[60], "max_inference_request_validity":[36000], "max_slowest_allowed_cadence":[36000]}'  > tx.json
```


not sure where to put this but that's how you do it from the command line

File: x/emissions/proto/emissions/state/v1/types.proto (hunk 1 above)
[the remark this answers is not in the document]
Comment (relyt29) [reply x/emissions/proto/emissions/state/v1/types.proto]: done in latest commit

**Review by guilherme-brandao — APPROVED** [review]
lgtm

PS: We prob want to add the `gov` module as a fast follow

File: x/emissions/module/autocli.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (kpeluso) [reply x/emissions/module/autocli.go:107]: in the external docs!

[the remark this answers is not in the document]
Comment (relyt29) [reply x/emissions/module/autocli.go:107]: added to external docs

**Pull request merged**